core/dbt/task/run.py

```diff
@@ -568,6 +568,9 @@ def _execute_microbatch_materialization(
                 # At least one batch has been inserted successfully!
                 incremental_batch = True
 
+            except (KeyboardInterrupt, SystemExit):
+                # reraise it for GraphRunnableTask.execute_nodes to handle
```

Comment:
Should we be writing out run results in this case?

Reply:
The KeyboardInterrupt handler in GraphRunnableTask.execute_nodes handles that